internal/evaluator/conftest_evaluator_test.go

```diff
@@ -1235,6 +1248,45 @@ func TestRuleMetadata(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "drop stale effectiveOn",
+			result: output.Result{
+				Metadata: map[string]any{
+					"code":        "warning2",
+					"collections": []any{"A"},
```

Comment:
Maybe this to be consistent with line 1264..?

```suggestion
					"collections": []string{"A"},
```